### ISSUE TITLE
ClayTreeView drag & drop into containers

### ISSUE THREAD

**Issue by boton** [liferay/clay#5213]
## Is there an example you can provide?
https://storybook.clayui.com/?path=/story/design-system-components-treeview--dynamic

I added two bugs on one because looks related I can split it if works better for you.

---

### What are the steps to reproduce?

- Drag something above the first child of a folder.
- We have two areas in a few pixels in the above one [^1]
- Drop in the above one

[^1]: Do we need the two areas in this case?

### What is the current result?

Move the item below the parent.

### What is the expected result?

Move the item inside the parent as the first child.

### Screen record

https://user-images.githubusercontent.com/67901/203999432-f18fd6c2-80d4-44b4-bcbb-a922aa7e13c9.mp4

---

### What are the steps to reproduce?

- Drag something below the last child of a folder.
- We have two areas in a few pixels in the above one. [^2]
- Drop in the above one

[^2]: It's difficult to perceive when the item is going to go inside the parent or after it.

### What is the current result?

Move the item inside the parent as the second-last child.

### What is the expected result?

Move the item inside the parent as the last child.

### Screen record

https://user-images.githubusercontent.com/67901/203998668-e5c18d56-14c8-4816-89a6-36ff6e9c4e62.mp4

I attached some videos for a better explanation, thanks to @AliciaGarciaGarcia for the screen recordings.

**Comment by matuzalemsteles** [comment]
Hey @boton thanks for reporting these two bugs, I'll leave a few more details:

> Do we need the two areas in this case?

We need both arias because the first is what places the element on the first level and the second is what places the element inside the child. I think the problem here is how to visually show it that it's positioning on the first level or the second level.

cc @drakonux this is an interesting case, what do you think we could do here? Can we maybe signal in a better way that the bar is on the first level or the second or when there is this use case where the node is expanded just move it into the field? in this case the user needs to close the node to move below it.

> It's difficult to perceive when the item is going to go inside the parent or after it

I totally agree, this really is a very strange bug and creates a lot of confusion for the user.

**Comment by boton** [comment]
> We need both arias because the first is what places the element on the first level and the second is what places the element inside the child. 

Maybe I missing something but between the parent and first children we don't need two areas only one is needed to put the dropped item in the first place.

If you want to put the item at the same level as the parent you need to drop it before or after the parent, not between the parent and the first child.

**Am I understanding it right? 
If I am missing something please could you explain it to me?**

**Comment by matuzalemsteles** [comment]
> Maybe I missing something but between the parent and first children we don't need two areas only one is needed to put the dropped item in the first place.
> If you want to put the item at the same level as the parent you need to drop it before or after the parent, not between the parent and the first child.

This is a great challenge for the TreeView, because if you want to move an element that is expanded above it or below it, you will have the same problem, so for the user to be able to move the element naturally, he would have to close the node to to achieve in a smooth manner without causing confusion as to which line is moving into the child or how to sibling.

The issue here is that we could even remove this top line but we will still have the line below the expanded element, we can also remove it but we will be limiting the user's possibilities to be able to move the sibling element of the expanded element.

I think the best thing here is to create a line differentiation for the user to understand that they are moving into the expanded child or when they are moving as the expanded child's sibling in these situations. I'm not sure if this would also be ideal here, another alternative would be to remove the line that moves like sibling when the element is expanded and the last line of the expanded element would only work when the last element of the expanded child is intended to move down of the expanded element, so we can cover the possibility of moving the element like sibling below the expanded element and above.

To summarize the thoughts would be something like:

- Expand node
- Move the child node above the first child node of the expanded node
- Move the child node to the last child node of the expanded node
- Move the last child node of the expanded node to be sibling of the expanded node but below it

I think there are several interaction options here that allow a better flow for the user, but I would like to know how to see the vision and thoughts about UX from @drakonux and @emiliano-cicero

**Comment by boton** [comment]
@matuzalemsteles thank you for the explanation. 

> This is a great challenge for the TreeView, because if you want to move an element that is expanded above it or below it, you will have the same problem, so for the user to be able to move the element naturally,

I agree that is a great challenge but as a user, the current behavior is totally unexpected. Dropping an item between the parent and the first child is going to add the item below the parent 🤯 

https://user-images.githubusercontent.com/67901/203999432-f18fd6c2-80d4-44b4-bcbb-a922aa7e13c9.mp4


> I think there are several interaction options here that allow a better flow for the user, but I would like to know how to see the vision and thoughts about UX from @drakonux and @emiliano-cicero

Totally agree

**Comment by matuzalemsteles** [comment]
Thanks @boton and @emiliano-cicero for all the details! I agree on the points, @emiliano-cicero I'll open a jira ticket to deal with it, thanks!

**Comment by boton** [comment]
Thank you both @emiliano-cicero @matuzalemsteles ❤️
